waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -93,6 +97,16 @@ private void add(AbstractMetaStore metaStore) {
     mappingsByPrefix.put(metaStoreMapping.getDatabasePrefix(), databaseMapping);
     Whitelist mappedDbWhitelist = new Whitelist(metaStore.getMappedDatabases());
     mappedDbByPrefix.put(metaStoreMapping.getDatabasePrefix(), mappedDbWhitelist);
+
+    List<MappedTables> mappedTables = metaStore.getMappedTables();
+    if (mappedTables != null) {
```

Comment:
Could this ever be an empty list? If so would it be OK to add an empty map on line 108?

Reply:
Yes, this could be an empty list and in that case `mappedTblByDb` stays empty and is added on line 108, so I think that's already happening.

Reply:
OK, as long as that's not going to cause any issues then I guess it's fine! Worth checking this path happens in at least one of the tests.

Reply:
Ok, added 1 test